DataProcessing/BybitFundingRateDownloader.cs

```diff
@@ -0,0 +1,333 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using QuantConnect;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+
+namespace QuantConnect.DataProcessing;
+
+/// <summary>
+/// BybitFundingRateDownloader implementation
+/// </summary>
+public class BybitFundingRateDownloader : IDisposable
+{
+    private const string BybitApiEndpoint = "https://api.bybit.com";
+    
+    private static readonly JsonSerializerSettings SerializerSettings = new JsonSerializerSettings
+    {
+        ContractResolver = new CamelCasePropertyNamesContractResolver()
+    };
+    
+    private readonly DateTime? _deploymentDate;
+    private readonly string _destinationFolder;
+    private readonly string _existingInDataFolder;
+
+    /// <summary>
+    /// Control the rate of download per unit of time.
+    /// </summary>
+    private readonly RateGate _indexGate = new(10, TimeSpan.FromSeconds(1));
+
+    private readonly BybitInstrumentInfo[] _perpetualFuturesExchangeInfos;
+
+    /// <summary>
+    /// Creates a new instance of <see cref="BybitFundingRateDownloader"/>
+    /// </summary>
+    /// <param name="destinationFolder">The folder where the data will be saved</param>
+    /// <param name="deploymentDate"></param>
+    public BybitFundingRateDownloader(string destinationFolder, DateTime? deploymentDate)
+    {
+        _deploymentDate = deploymentDate;
+        _destinationFolder = Path.Combine(destinationFolder, "cryptofuture", "bybit", "margin_interest");
+        _existingInDataFolder = Path.Combine(Globals.DataFolder, "cryptofuture", "bybit", "margin_interest");
+
+        Directory.CreateDirectory(_destinationFolder);
+
+        _perpetualFuturesExchangeInfos = GetExchangeInfo()
+            .Where(x => x.ContractType.EndsWith("Perpetual", StringComparison.InvariantCultureIgnoreCase))
+            .ToArray();
+    }
+
+
+    /// <summary>
+    /// Runs the instance of the object.
+    /// </summary>
+    /// <returns>True if process all downloads successfully</returns>
+    public bool Run()
+    {
+        var ratePerSymbol = new Dictionary<string, Dictionary<DateTime, decimal>>();
+        foreach (var date in GetProcessingDates())
+        {
+            foreach (var apiFundingRate in GetData(date))
```

Comment:
The process can be sped up immensely by fetching the data for multiple days. I can add this if it's desired.

Reply:
I think as is it's okay 👍  the data jobs will run with the single `_deploymentDate` set